gateway/reverse_proxy.go

```diff
@@ -403,13 +403,19 @@ func (p *ReverseProxy) defaultTransport(dialerTimeout float64) *http.Transport {
 		dialContextFunc = p.Gw.dnsCacheManager.WrapDialer(dialer)
 	}
 
-	return &http.Transport{
+	transport := &http.Transport{
 		DialContext:           dialContextFunc,
 		MaxIdleConns:          p.Gw.GetConfig().MaxIdleConns,
 		MaxIdleConnsPerHost:   p.Gw.GetConfig().MaxIdleConnsPerHost, // default is 100
 		ResponseHeaderTimeout: time.Duration(dialerTimeout) * time.Second,
 		TLSHandshakeTimeout:   10 * time.Second,
 	}
+
+	if p.Gw.dialCtxFn != nil {
+		transport.DialContext = p.Gw.dialCtxFn
```

Comment:
Would you not override this sooner (L407, L403...?)

Reply:
yep, that'd do :)